packages/gatsby/src/utils/jobs-manager.ts

```diff
@@ -39,48 +50,60 @@ const convertPathsToAbsolute = filePath => {
 }
 /**
  * Get contenthash of a file
- *
- * @param {string} path
  */
-const createFileHash = path => hasha.fromFileSync(path, { algorithm: `sha1` })
+function createFileHash(path: string): string {
+  return hasha.fromFileSync(path, { algorithm: `sha1` })
+}
 
-/**
- * @typedef BaseJobInterface
- * @property {string} name
- * @property {string} outputDir,
- * @property {Record<string, any>} args
-
- * @typedef JobInputInterface
- * @property {string[]} inputPaths
- * @property {{name: string, version: string, resolve: string}} plugin
-
- * @typedef InternalJobInterface
- * @property {string} id
- * @property {string} contentDigest
- * @property {{path: string, contentDigest: string}[]} inputPaths
- * @property {{name: string, version: string, resolve: string, isLocal: boolean}} plugin
- *
- * I know this sucks but this is the only way to do it properly in jsdoc..
- * @typedef {BaseJobInterface & JobInputInterface} JobInput
- * @typedef {BaseJobInterface & InternalJobInterface} InternalJob
- */
+interface IBaseJob {
+  name: string
+  outputDir: string
+  args: Record<string, any>
+}
+
+interface IJobInput {
+  inputPaths: string[]
+  plugin: {
+    name: string
+    version: string
+    resolve: string
+  }
+}
+
+interface IInternalJob {
```

Comment:
It's a little next level, but if this had a generic param then you could type the return type of the job and prevent the `Promise<object>` occurrences, but rather give them explicit (even by parametric) types. I'm not sure how much work it would be to get that in, though. So I'm not asking for this change, just something to consider.

Reply:
I think that would be a nice change to make, just had a look at it, but I wasn't able to make that change quickly. I think that would be nice to do in an entirely different PR.